arangod/RocksDBEngine/RocksDBGeoIndex.cpp

```diff
@@ -762,9 +776,9 @@ std::unique_ptr<IndexIterator> RocksDBGeoIndex::iteratorForCondition(
   if (!params.sorted &&
       (params.filterType == geo::FilterType::CONTAINS ||
        params.filterType == geo::FilterType::INTERSECTS) &&
-      (params.minDistanceRad() < geo::kRadEps &&
-       params.maxDistanceRad() >
-           geo::kMaxRadiansBetweenPoints - geo::kRadEps)) {
+      (params.minDistance <= geo::kMetersEps &&
+       params.maxDistance >=
+           geo::kMaxDistanceBetweenPoints - geo::kMetersEps)) {
```

Comment:
What about test like:

```
LET P = ...
FOR d in ... FILTER GEO_DISTANCE(p, d) > 0
```

I mean what if user don't want same points, only distinct?
I still think here will be better to have boolean condition

Reply:
OK, I will try to think of something...

Reply:
This is indeed a bug, as it seems. Because even with `GEO_DISTANCE(p, d) >= 0` (which should return the point), it is not returned. Looking into it...